> This may be an issue with Jinja -- FileSystemLoader calls `os.fspath`, but perhaps it should also call `os.path.normpath`.
> 
> https://github.com/pallets/jinja/blob/c3a61d6ef654f389ea2bdeddce0ffc74d656be8b/src/jinja2/loaders.py#L188
> 
> A

this bug doesn't seem to be related to Jinja -- FileSystemLoader from the call stack：

```

Traceback (most recent call last):
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\builders\html\__init__.py", line 1048, in handle_page
    output = self.templates.render(templatename, ctx)
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\jinja2glue.py", line 188, in render
    return self.environment.get_template(template).render(context)
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__jinja2\jinja2\environment.py", line 1291, in render
    self.environment.handle_exception()
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__jinja2\jinja2\environment.py", line 926, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__furo\furo\theme\furo\page.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__furo\furo\theme\furo\base.html", line 57, in top-level template code
    {%- block styles -%}
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__furo\furo\theme\furo\base.html", line 71, in block 'styles'
    {%- block theme_styles %}
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__furo\furo\theme\furo\base.html", line 72, in block 'theme_styles'
    {% include "partials/_head_css_variables.html" with context %}
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\jinja2glue.py", line 209, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: partials/_head_css_variables.html

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\cmd\build.py", line 276, in build_main
    app.build(args.force_all, filenames)
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\application.py", line 330, in build
    self.builder.build_update()
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\builders\__init__.py", line 286, in build_update
    self.build(to_build,
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\builders\__init__.py", line 350, in build
    self.write(docnames, list(updated_docnames), method)
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\builders\__init__.py", line 524, in write
    self._write_serial(sorted(docnames))
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\builders\__init__.py", line 534, in _write_serial
    self.write_doc(docname, doctree)
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\builders\html\__init__.py", line 625, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "\\?\C:\Users\dhmem\AppData\Local\Temp\Bazel.runfiles_uu7w7nnu\runfiles\aidi_sdk_requirements\pypi__sphinx\sphinx\builders\html\__init__.py", line 1055, in handle_page
    raise ThemeError(__("An error happened in rendering the page %s.\nReason: %r") %
sphinx.errors.ThemeError: An error happened in rendering the page advanced_topics.
Reason: TemplateNotFound('partials/_head_css_variables.html')

Theme error:
An error happened in rendering the page advanced_topics.
Reason: TemplateNotFound('partials/_head_css_variables.html')
Target //documents:documents failed to build
```

And It seems that `os.path.normpath` has no effect on extended-length path:
```
>>> filename= "C:\\Windows\\any/path"
>>> os.path.normpath(filename)
'C:\\Windows\\any\\path'
>>> ext_path = "\\\\?\\C:\\Windows\\any/path"
>>> os.path.normpath(ext_path)
'\\\\?\\C:\\Windows\\any/path'
```

--- a/sphinx/jinja2glue.py
+++ b/sphinx/jinja2glue.py
@@ -1,7 +1,7 @@
 """Glue code for the jinja2 templating engine."""
 
+import pathlib
 from os import path
-import os
 from pprint import pformat
 from typing import TYPE_CHECKING, Any, Callable, Dict, Iterator, List, Tuple, Union
 
@@ -117,10 +117,7 @@ class SphinxFileSystemLoader(FileSystemLoader):
 
     def get_source(self, environment: Environment, template: str) -> Tuple[str, str, Callable]:
         for searchpath in self.searchpath:
-            filename = path.join(searchpath, template)
-            # Repalce '/' by os.sep if The searchpath is an extended-length path.
-            if filename.startswith("\\\\?\\"):
-                filename = filename.replace('/', os.sep)
+            filename = str(pathlib.Path(searchpath, template))
             f = open_if_exists(filename)
             if f is None:
                 continue

